src/main.rs

```diff
@@ -1,262 +1,4 @@
-                received_notifications: notification_counter,
-            },
-        )
-    } else {
-        // Only log to console if telemetry disabled
-        tracing_subscriber::fmt()
-            .with_max_level(state.config.log_level())
-            .with_span_events(FmtSpan::CLOSE)
-            .init();
-    }
-
-    let port = state.config.port;
-    let build_version = state.build_info.crate_info.version.clone();
-    let build_commit = state
-        .build_info
-        .version_control
-        .clone()
-        .unwrap()
-        .git()
-        .unwrap()
-        .commit_short_id
-        .clone();
-    let build_rustc_version = state.build_info.compiler.version.clone();
-
-    let state_arc = Arc::new(state);
-
-    let global_middleware = ServiceBuilder::new().layer(
-        TraceLayer::new_for_http()
-            .make_span_with(DefaultMakeSpan::new().include_headers(true))
-            .on_request(DefaultOnRequest::new().level(Level::INFO))
-            .on_response(
-                DefaultOnResponse::new()
-                    .level(Level::INFO)
-                    .include_headers(true),
-            ),
-    );
-
-    let app = Router::with_state(state_arc)
-        .route("/health", get(handlers::health::handler))
-        .route(
-            "/clients",
-            post(handlers::single_tenant_wrappers::register_handler),
-        )
-        .route(
-            "/clients/:id",
-            delete(handlers::single_tenant_wrappers::delete_handler),
-        )
-        .route(
-            "/clients/:id",
-            post(handlers::single_tenant_wrappers::push_handler),
-        )
-        .route(
-            "/:tenant_id/clients",
-            post(handlers::register_client::handler),
-        )
-        .route(
-            "/:tenant_id/clients/:id",
-            delete(handlers::delete_client::handler),
-        )
-        .route(
-            "/:tenant_id/clients/:id",
-            post(handlers::push_message::handler),
-        )
-        .layer(global_middleware);
-
-    let header = format!(
-        "
- ______       _               _____
-|  ____|     | |             / ____|
-| |__    ___ | |__    ___   | (___    ___  _ __ __   __ ___  _ __
-|  __|  / __|| '_ \\  / _ \\   \\___ \\  / _ \\| '__|\\ \\ / // _ \\| '__|
-| |____| (__ | | | || (_) |  ____) ||  __/| |    \\ V /|  __/| |
-|______|\\___||_| |_| \\___/  |_____/  \\___||_|     \\_/  \\___||_|
-\nversion: {}, commit: {}, rustc: {},
-web-host: {}, web-port: {},
-providers: [{}]
-",
-        build_version,
-        build_commit,
-        build_rustc_version,
-        "0.0.0.0",
-        port.clone(),
-        supported_providers_string
-    );
-    println!("{}", header);
-
-    let addr = SocketAddr::from(([0, 0, 0, 0], port));
-    axum::Server::bind(&addr)
-        .serve(app.into_make_service())
-        .await
-        .unwrap();
-
-    Ok(())
+async fn main() -> echo_server::error::Result<()> {
+    Ok(echo_server::bootstap().await?)
```

Comment:
```suggestion
    dotenv().ok();
    let config = env::get_config()
        .expect("Failed to load config, please ensure all env vars are defined.");
    Ok(echo_server::bootstap(config).await?)
```

Pass config into `bootstrap(...)` so that in tests you can specify the config